newm-tx-builder/src/main/kotlin/io/newm/txbuilder/ktx/NativeAssetKtx.kt

```diff
@@ -38,6 +40,29 @@ fun List<NativeAsset>.toNativeAssetCborMap(): CborMap? {
     )
 }
 
+/**
+ * Convert into a List where amounts of the same asset type are summed.
+ */
+fun List<NativeAsset>.mergeAmounts(): List<NativeAsset> {
+    val map = mutableMapOf<String, Pair<NativeAsset, BigInteger>>()
+    for (asset in this) {
+        val key = asset.policy + asset.name
+        var amount = asset.amount.toBigInteger()
```

Comment:
@AndrewWestberg I'm assuming that we should be using BigInteger here for amount. If this is an overkill let me know and I can change to Long